src/trg-client.c

```diff
@@ -620,23 +690,25 @@ trg_http_perform_inner(TrgClient * tc, trg_request * request,
     response->size = 0;
     response->raw = NULL;
 
-    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, request->body);
-    curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *) response);
+    curl_easy_setopt (curl, CURLOPT_POSTFIELDS, request->body);
+    curl_easy_setopt (curl, CURLOPT_WRITEDATA, (void *) response);
 
-	session_id = trg_client_get_session_id(tc);
-	if (session_id)
-		headers = curl_slist_append(NULL, session_id);
+    session_id = trg_client_get_session_id (tc);
+    if (session_id)
+      headers = curl_slist_append (NULL, session_id);
 
-	curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+    trg_client_inject_custom_headers (tc, &headers);
 
-    response->status = curl_easy_perform(curl);
+    curl_easy_setopt (curl, CURLOPT_HTTPHEADER, headers);
+
+    response->status = curl_easy_perform (curl);
 
     curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &httpCode);
 
-    g_free(session_id);
 
     if (headers)
-    	curl_slist_free_all(headers);
+        trg_http_headers_free(headers);
+//    	curl_slist_free_all(headers);
```

Comment:
?

Reply:
I wanted to ask about that, the function `trg_http_headers_free` goes over all items and g_free them, but if I call `curl_slist_free_all` it segfaults, should I call something else? I had a feeling that it would just deallocate curl_slist's struct but it seems it double freed data.

Reply:
`curl_slist_free_all()` looks like the correct function. I'd suggest running under `gdb` to inspect why it crashes.

According to the docs `curl_slist_append()` always creates a copy of the string its given.